### DISCUSSION TITLE
Discord Webhook?

### DISCUSSION THREAD

**Discussion by joryirving** [itzg/docker-minecraft-server#788] [Ideas]
Hi there, was wondering if there was a possible way to integrate with discord via webhook? I was going to look into mangling a bot together or something using something like this in an attached container:
https://github.com/destruc7i0n/shulker

I also found a possibly abandoned project that tried to wrap it into the same container
:https://github.com/pcjpnet/docker_minecraft_discord_webhook/blob/master/Dockerfile

But I was mostly inspired by the valheim image I'm using has it baked directly into the docker image where you only need to pass a webhook URL into it via an environment variable.
https://github.com/mbround18/valheim-docker/pull/158

I'm just wondering, before I try to reinvent the wheel, if anyone else has done this, or can point me in the proper direction.

**Comment by itzg** [comment]
I haven't heard of anyone doing that, but very interesting idea. It seems like it would be more efficient to run/implement that as a plugin, such as

https://www.spigotmc.org/resources/discordsrv.18494/

since the solutions you linked are a little brittle IMHO since they have to tap in via rcon or stdio.

**Reply by joryirving** [reply]
As a workaround, I converted the vanilla server to forge, and used this method:
https://www.curseforge.com/minecraft/mc-mods/discord-integration
![image](https://user-images.githubusercontent.com/46251616/109526865-4a83a780-7a70-11eb-9b5f-f46577712fb7.png)

**Reply by Bmiest** [reply]
I use the same valheim image and the discord webhook is great. Would love to see this implemented the same way!

**Reply by Geekmeleon** [reply]
Did everyone need to run a forge version instead of vanilla then?

**Reply by itzg** [reply]
> Did everyone need to run a forge version instead of vanilla then?

Well, you can't apply mods/plugins to vanilla, so I'm not sure how you'd add Discord support otherwise.

**Comment by Geekmeleon** [comment]
This guys Valheim server I use, has discord integration using webhooks and log file reading.  Maybe seeing his project can help integrate this. 
https://hub.docker.com/r/lloesche/valheim-server

**Comment by bagobones** [comment]
Probably should be a separate project but I really like the idea of something like shulker that both acts as a chat bridge using Logs and RCON and allows specific users to send some commands over discord (however there are risks with that). 

I currently handle messaging with a number of separate solutions, I have may backup scripts call RCON to announce to players, I have the Minecraft Home Assistant integration send player count / online offline notices to discord and I have some CRON jobs that send RCON announcements for over viewer updates etc.

In general having something broad like webhook support etc to announce health / restarts issues etc could be something that fits and would allow for broad integration.

Riskier things like allowing admin commands over chat or webhook would need to be carefully thought out and optional to enable / dissable.
